### PR TITLE
feat(Module:Drawer) auto focus input on drawer open

### PR DESCRIPTION
**Pull request by eizzn**
when a drawer opens and it contains a form, the first visible input gains focus

### 🤔 This is a ...

- [ ] New feature
- [ ] Bug fix
- [ ] Site / documentation update
- [ ] Demo update
- [ ] Component style update
- [ ] Bundle size optimization
- [ ] Performance optimization
- [ ] Refactoring
- [ ] Code style optimization
- [ ] Test Case
- [ ] Branch merge
- [ ] Other (about what?)

### 📝 Changelog

| Language   | Changelog |
| ---------- | --------- |
| 🇺🇸 English |           |
| 🇨🇳 Chinese |           |

### ☑️ Self Check before Merge

⚠️ Please check all items below before review. ⚠️

- [ ] Doc is updated/provided or not needed
- [ ] Demo is updated/provided or not needed
- [ ] Changelog is provided or not needed

### CODE REVIEW

File: components/core/JsInterop/modules/dom/manipulationHelper.ts [hunk 1]
```diff
@@ -242,6 +242,31 @@ export class manipulationHelper {
...
+          let forms = parent.getElementsByTagName('form');
+          if (forms.length > 0) {
+              let form = forms[0];
+              let inputs = form.querySelectorAll('input, textarea, select');
+              try {
+                  [].forEach.call(inputs, function (input) {
+                      var style = window.getComputedStyle(input);
+                      if (style.display !== 'none' && style.visibility !== 'hidden') {
+                          input.focus();
+                          throw BreakException;
```
Comment (ElderJames) [inline components/core/JsInterop/modules/dom/manipulationHelper.ts:258]: Why not just use `foreach ... of` loop?

File: components/drawer/Drawer.razor.cs [hunk 2]
```diff
@@ -441,6 +441,7 @@ protected override async Task OnAfterRenderAsync(bool isFirst)
                     }
             }
             await base.OnAfterRenderAsync(isFirst);
+            await JsInvokeAsync(JSInteropConstants.DomMainpulationHelper.FocusOnChildFormIfExists, Id);
```
Comment (ElderJames) [inline components/drawer/Drawer.razor.cs:444]: Maybe this should be call every time the drawer was opened.

**Review by ElderJames — COMMENTED** [review]
Thank you for the contribution @eizzn . Please add a new parameter to perform the feature because there are not all drawer are including form.

**Comment by eizzn** [comment]
did not go with parameter because the js checks if there is a form before attempting focus.

File: components/core/JsInterop/modules/dom/manipulationHelper.ts [hunk 3]
```diff
@@ -242,6 +242,31 @@ export class manipulationHelper {
...
+          let forms = parent.getElementsByTagName('form');
+          if (forms.length > 0) {
+              let form = forms[0];
+              let inputs = form.querySelectorAll('input, textarea, select');
+              try {
+                  inputs.forEach(function (input:HTMLElement) {
+                      var style = window.getComputedStyle(input);
+                      if (style.display !== 'none' && style.visibility !== 'hidden') {
+                          input.focus();
+                          throw BreakException;
```
Comment (ElderJames) [inline components/core/JsInterop/modules/dom/manipulationHelper.ts:258]: I don't think throwing an exception is a good way to exit the loop. You can use `inputs.find(input=> ... )` to find the input

**Comment by eizzn** [comment]
closing
use HTML attribute autofocus as alternative

**Pull request closed without merging**